src/main/java/io/tiledb/java/api/Group.java

```diff
@@ -24,12 +24,403 @@
 
 package io.tiledb.java.api;
 
+import io.tiledb.libtiledb.SWIGTYPE_p_int;
+import io.tiledb.libtiledb.SWIGTYPE_p_p_char;
+import io.tiledb.libtiledb.SWIGTYPE_p_p_tiledb_config_t;
+import io.tiledb.libtiledb.SWIGTYPE_p_p_tiledb_group_t;
+import io.tiledb.libtiledb.SWIGTYPE_p_p_void;
+import io.tiledb.libtiledb.SWIGTYPE_p_tiledb_datatype_t;
+import io.tiledb.libtiledb.SWIGTYPE_p_tiledb_group_t;
+import io.tiledb.libtiledb.SWIGTYPE_p_tiledb_object_t;
+import io.tiledb.libtiledb.SWIGTYPE_p_unsigned_int;
+import io.tiledb.libtiledb.SWIGTYPE_p_unsigned_long_long;
 import io.tiledb.libtiledb.tiledb;
+import java.math.BigInteger;
 
-public class Group {
-
+public class Group implements AutoCloseable {
   private final Context ctx;
   private final String uri;
+  private SWIGTYPE_p_tiledb_group_t groupp;
+  private SWIGTYPE_p_p_tiledb_group_t grouppp;
+
+  public Group(Context ctx, String uri) throws TileDBError {
```

Comment:
Let's make the constructor take a third parameter for the mode, like the C++ API.

Reply:
All done, thanks for the comments!